Services/Sources/Services/LocationService/LocationManager.swift

```diff
@@ -0,0 +1,173 @@
+import Foundation
+import CoreLocation
+
+public typealias AuthorizationStatus = ((CLAuthorizationStatus) -> Void)
+public typealias LocationResponce = ((Result<CLLocation, Error>) -> Void)
+
+public protocol LocationManager {
```

Comment:
In my opinion, `LocationManager` isn't the best name for protocol, because it sounds like a class name. On top of that, it forces us to give worse names to the other classes (`LocationManagerImpl`). I would suggest to stick to this approach in protocol naming:

Performed by = name+ing
Performed on = name+able
Performed for = name+delegate
Unable to identify = name + protocol
[(from here)](https://santoshbotre01.medium.com/naming-a-protocol-in-swift-b0d933b44ddc)

According to it, the best choice will be `LocationManagerProtocol`